common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -272,28 +272,37 @@ void reclaimAllSpaceAndUnlink() {
                     head.next = null;
                     head = next;
                 }
-                if (reclaimSpace != 0) {
+                if (reclaimSpace > 0) {
                     reclaimSpace(reclaimSpace);
                 }
             }
 
-            void reclaimSpace(int space) {
-                assert space >= 0;
+            private void reclaimSpace(int space) {
                 availableSharedCapacity.addAndGet(space);
             }
 
-            boolean reserveSpace(int space) {
-                return reserveSpace(availableSharedCapacity, space);
+            void relink(Link link) {
+                if (this.link != link) {
```

Comment:
I don't think this check was there before, was it intentional?

If so then the `this.link = link;` below could be moved inside the if block?

Reply:
I think we can just remove it...